addons/web_editor/models/ir_ui_view.py

```diff
@@ -80,10 +86,26 @@ def save_oe_structure(self, el):
             'mode': 'extension',
         }
         vals.update(self._save_oe_structure_hook())
-        self.env['ir.ui.view'].create(vals)
+        oe_structure_view = self.env['ir.ui.view'].create(vals)
+        self._copy_custom_snippet_translations(oe_structure_view, 'arch')
 
         return True
 
+    @api.model
+    def _copy_custom_snippet_translations(self, record, html_field):
+        """ Given a ``record`` and its HTML ``field``, detect any
+        usage of a custom snippet and copy its translations.
+        """
+        tree = html.fromstring(record[html_field])
+        for custom_snippet_el in tree.xpath('//*[hasclass("s_custom_snippet")]'):
+            custom_snippet_name = custom_snippet_el.get('data-name')
+            custom_snippet_view = self.search([('name', '=', custom_snippet_name)], limit=1)
+            if custom_snippet_view:
+                # TODO: use the `excluded` field param of `copy_translations`
```

Comment:
@HydrionBurst As discussed, here I ideally should be able to tell to the `copy_translations()` (or any new method) which field's translations from model A I want to copy into another model B's field translations.

Like being able to copy the translations of an `ir.ui.view.arch_db` into a `product.template.product_description`.